code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/templates/MockAEPPushTemplateDataProvider.kt

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.notificationbuilder.internal.templates
+
+import android.os.Bundle
+import com.adobe.marketing.mobile.notificationbuilder.PushTemplateIntentConstants
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateConstants
+import org.mockito.Mockito
+import org.mockito.kotlin.mock
+
+object MockAEPPushTemplateDataProvider {
+    fun getMockedDataMapWithBasicData(): HashMap<String, String> {
```

Comment:
nit: can this be returned as a `Map` rather than `HashMap`? our push template classes accept map objects so i don't see a need to use a `HashMap` here

Reply:
This is now being kept as MutableMap, which will provide functionality to remove / replace any keys in existing data, for any specific test requirement.
This Mutable map is then passed as parameter to create push template. Kotlin takes care of the conversion from Mutablemap to Map